docs/branching.md

```diff
@@ -0,0 +1,32 @@
+### Branching and deployments model
+
+This document aims to discuss and document the relations between branches and deployments of `elixir-omg`, with respect to branches and deployments of `plasma-contracts`.
+
+It is a refinement of the [OIP4 branching model](https://github.com/omisego/OIP/blob/master/0004-ewallet-release-and-versioning.md), applicable to `elixir-omg` and `plasma-contracts` versioning.
+
+This takes `v0.1` and `v0.2` as an example of two versions, where one is deployed in `staging`/`Ari` and the other is the upcoming upgrade and is deployed in `devel`.
+
+Rationale:
+- the history and the relations between the versions are readable and accountable,
+- we can predictably sync our respective watchers/run child chain servers against deployed contracts
+- and at the same time be able to move with `master`s quickly
+
+("->" means that the `mix.exs` holds the following branch)
+
+- `elixir-omg/v0.1` -> `plasma-contracts/v0.1`, that's understood, the `mix.lock` points to a specific frozen version, no changes expected there
+- `elixir-omg/master` -> `plasma-contracts/master`, mix lock points to the contracts ~deployed on `devel`~ we want to CI-check with `elixir-omg`. So it might be newer than what contract is deployed on `devel`
+- `elixir-omg/v0.2` -> `plasma-contracts/v0.2`, the commit that fixes *the branch `plasma-contracts/v0.2` in `elixir-omg`* will need to be constantly rebased onto `master`, as long we keep looping features into `elixir-omg/v0.2`. *The mix lock will always point to the contract commit currently deployed on `devel`*.
```

Comment:
Rebasing changes on release branch onto master may cause a lot of grief, since we'll be replaying the changes on a moving base. It will be easier to simply merge.

What do you mean by "looping features into elixir-omg/v0.2"? From master?

Reply:
That language works better, thanks.